tmt/utils.py

```diff
@@ -671,6 +686,21 @@ def shell_variables(data):
     return [f"{key}={shlex.quote(str(value))}" for key, value in data.items()]
 
 
+def duration_to_seconds(duration):
+    """ Convert sleep time format into seconds """
+    units = {
+        's': 1,
+        'm': 60,
+        'h': 60 * 60,
+        'd': 60 * 60 * 24,
+        }
+    for unit in units:
+        matched = re.search(f"^(\d+){unit}$", duration)
+        if matched:
+            return int(matched.group(1)) * units[unit]
```

Comment:
IMO we don't need the for cycle. Also since we are using sleep time we should honour it's default being 'seconds' (see man sleep) - which now raises Exception

```suggestion
   try:
       number, suffix = re.match(r'^(\d+)([smhd]?)$', duration).groups()
       return int(number) * units.get(suffix, 1)
   except (ValueError, AttributeError) as err:
       pass
```

Reply:
Oh, that's nicely elegant! Thanks, @lukaszachy. Applied and added a simple unit test as well.